### PR TITLE
ceph-pr-docs: add annotation to build result

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: ceph-pr-docs/build/build [hunk 1]
```diff
@@ -12,4 +12,34 @@ sudo rm -f /etc/apt/sources.list.d/ceph-boost*
 sudo apt-get update -o Acquire::Languages=none -o Acquire::Translation=none
 sudo apt-get install -y gcc python3-dev python3-pip python3-virtualenv libxml2-dev libxslt-dev doxygen graphviz ant ditaa cython3
 
-timeout 3600 ./admin/build-doc
+virtualenv -q --python python3 venv
+. venv/bin/activate
+pip install tox
+pip install git+https://github.com/ceph/githubcheck.git
+sha1=$(git rev-parse refs/remotes/origin/pr/${ghprbPullId}/head)
```
Comment (djgalloway) [inline ceph-pr-docs/build/build:19]: Since this is the second time I've seen this block, may I suggest it go in `build_utils.sh` instead?

Comment (tchaikov) [reply ceph-pr-docs/build/build:19]: @djgalloway can we do this in another PR? after a second thought, i think we need to fix https://github.com/ceph/ceph-build/blob/01968d02ab54b07eae57c357362d0b8a2e44e730/scripts/build_utils.sh#L5-L6 first. it *always* creates a temp directory every time we include this script. and it is never removed.

Comment (epuertat) [inline ceph-pr-docs/build/build:19]: [`GIT_COMMIT`](https://plugins.jenkins.io/git/#environment-variables)?

File: ceph-pr-docs/build/build [hunk 2]
```diff
@@ -12,4 +12,34 @@ sudo rm -f /etc/apt/sources.list.d/ceph-boost*
...
+
+output=$(mktemp $PWD/build-doc-XXX.out)
+
+if timeout 3600 ./admin/build-doc 2>&1 | tee ${output}; then
```
Comment (epuertat) [inline ceph-pr-docs/build/build:23]: If build-doc fails there, wouldn't `pipefail` flag interrupt the script?

File: ceph-pr-docs/build/build (hunk 1 above)
Comment (tchaikov) [reply ceph-pr-docs/build/build:19]: i think `GIT_COMMIT` is the *merge* commit, not the one of the sha1 of the change itself. and it somehow fails to serve us sometimes. see 74d8f325149969a7527355c9df2c40b62c354cf4 .

File: ceph-pr-docs/build/build (hunk 2 above)
Comment (tchaikov) [reply ceph-pr-docs/build/build:23]: @epuertat i updated the commit message to explain the changes made in this commit. could you take another look?

File: ceph-pr-docs/build/build [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 
-set -ex
+set -xo pipefail
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply ceph-pr-docs/build/build:3]: i put more details in the commit message to explain why "set -e" should be dropped.

File: ceph-pr-docs/build/build (hunk 1 above)
Comment (epuertat) [reply ceph-pr-docs/build/build:19]: Weird :s I just saw a few related bugs on Jenkins Jira...

File: ceph-pr-docs/build/build (hunk 2 above)
Comment (epuertat) [reply ceph-pr-docs/build/build:23]: Sure.

File: ceph-pr-docs/build/build (hunk 3 above)
[the remark this answers is not in the document]
Comment (epuertat) [reply ceph-pr-docs/build/build:3]: Ok, understood now. You could have also used `trap ... ERR` as an error handler?

**Review by epuertat — APPROVED** [review]
Looks good! Do we have any sample PR where this ran?

File: ceph-pr-docs/build/build (hunk 3 above)
[the remark this answers is not in the document]
Comment (tchaikov) [reply ceph-pr-docs/build/build:3]: yes, but i think it's more straightforward and more explicit to use a variable than using a trap.

**Pull request merged**